---
title: "Interactome reconstruction from co-elution data with PrInCE"
author:
- name: Michael Skinnider
  affiliation: Michael Smith Laboratories, University of British Columbia, 
    Vancouver, Canada
  email: michael.skinnider@msl.ubc.ca
package: PrInCE
abstract: >
  Co-elution proteomics are a family of high-throughput methods to map protein-protein 
  interaction networks and their dynamics in cellular stimulation or differentiation. 
  These methods, also referred to as co-migration, co-fractionation, or
  protein correlation profiling, involve separating interacting protein complexes on the 
  basis of their diameter or biochemical properties. Protein-protein 
  interactions can then be inferred for pairs of proteins with similar 
  elution profiles. PrInCE implements a machine-learning approach to identify 
  protein-protein interactions given a set of labelled examples, using features
  derived exclusively from the data. This allows PrInCE to infer high-quality 
  protein interaction networks from raw proteomics data, without bias towards 
  known interactions or functionally associated proteins, 
  making PrInCE a unique computational resource for discovery. We provide an 
  overview of the key functionalities of the PrInCE R package, and demonstrate 
  an example of the analysis of data from a co-elution experiment investigating 
  the response of the cytoplasmic interactome to Fas-mediated apoptosis. 
output: 
  BiocStyle::html_document:
    toc_float: true
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
bibliography: bibliography.bib
csl: pnas.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction: What is PrInCE?

Proteins are 
Yet cellular functions are rarely accomplished by single proteins, acting in isolation.
Instead, most functions are accomplished by 


- what are protein-protein networks
- how they were historically mapped
- what is co-migration

FIGURE: heatmap, complex size going from large to small, gene names on axis

- what does PrInCE do
There are a few things that make PrInCE unique. 
- gaussians; 

## Installation

## Example: Interactome rearrangements in apoptosis

To demonstrate the use of PrInCE, we will work through a small example that is derived from Scott _et al._, 2017 [@scott2017]. 
In this paper, the authors mapped rearrangements in the cytoplasmic and membrane interactome during Fas-mediated apoptosis.
Control and stimulated cytoplasmic and membrane interactomes were quantified in three replicates each, meaning the complete dataset consists of twelve replicates. 
In practice, each set of replicates would be analyzed together (for a total of four networks).
However, such a complete analysis of the dataset would take over an hour, so for this vignette we focus on a single replicate.
The replicate in question is the first cytoplasmic replicate from the Fas-stimulated condition, and is bundled with the PrInCE package; it can be loaded with the following command:

```{r}
library(PrInCE)
data(scott)
```

The dataset consists of ratiometric protein quantitations, achieved by SILAC (stable isotope labelling by amino acids in cell culture), for 1,560 proteins in 55 size exclusion chromatography (SEC) fractions:

```{r}
dim(scott)
```

Each protein was quantified in at least one fraction; however, many measurements are missing:

```{r}
scott[1:10, 1:5]
```

To predict protein-protein interactions using PrInCE's machine-learning approach, we also need two pieces of information to train the classifier: a set of true positive interactions, and a set of true negative interactions.
In practice, we recommend providing a list of experimentally verified protein complexes; PrInCE assumes intra-complex interactions represent true positives, and inter-complex interactions represent true negatives.
These can be obtained from a number of sources, such as the CORUM database [@giurgiu2018], or our own subset of CORUM that removes complexes which may not always co-elute [@stacey2018].
We provide a third option with the PrInCE R package due to its CC-BY license, consisting of a list of 477 human protein complexes Complex Portal [@meldal2018]. 

```{r}
data(gold_standard)
head(gold_standard)
```

### Predicting protein-protein interactions: default analysis

The main function of the PrInCE package, `PrInCE::PrInCE`, provides an end-to-end workflow for predicting protein-protein interaction networks from the raw co-elution data. 
Briefly, this function first filters proteins with too little information to permit data analysis, then cleans the profiles for the remaining proteins and fits a mixture of Gaussians to each cleaned profile. 
PrInCE then calculates six features for each protein pair, from either the raw profiles, the cleaned profiles, or the fitted Gaussian mixture models, and concatenates features across replicates if more than one replicate was used.
These features are used as input to a machine learning model, along with the set of 'gold standard' true positive and true negative interactions, which uses a ten-fold cross-validation procedure to assign scores to each protein pair.
Protein pairs are ranked by their classifier scores and the precision at each point in the ranked list is calculated.
The entire list is returned to a user, who can select a precision threshold that matches their needs.



### Predicting protein-protein interactions: detailed analysis

Users who wish to 

## Session info

```{r}
sessionInfo()
```

## References

Â 